lib/sync_engine/2020_01_15/user_manager.rb

```diff
@@ -12,25 +12,28 @@ def change_pw(user, password, params)
 
         # We want to create a new session only if upgrading from a protocol version that does not
         # support sessions (i.e: 003 to 004).
-        if upgrading_protocol_version
-          handle_auth_response(user, params)
+        if upgrading_protocol_version && new_protocol_version == @user_class::SESSIONS_PROTOCOL_VERSION
+          handle_successful_authentication(user, params)
```

Comment:
The naming of this function still doesn't lend itself to being self evident. When reading this code, it reads awkwardly:

"if upgrading protocol version and new protocol version is 004, then handle successful authentication". See what I mean? It needs a comment to describe why. But in that case that comment would be a code smell.

Can we rename `handle_successful_authentication` to something more fitting. It needs to start with `create` so that we understand something new is being created, whether jwt or session. 

How about we have two functions:

`create_jwt`
`create_session`

These can be called directly, but they are also called by `handle_successful_authentication`.

Reply:
`handle_successful_authentication` is a method that I need to override in `20200115`.

If not, I would have to redefine the `sign_in`, `register`, `change_pw`, and `update` methods. And call the `create_session` method on successful authentication.

Reply:
I agree that `handle_successful_authentication` doesn't seem to be self evident.